CMakeLists.txt

```diff
@@ -459,38 +454,25 @@ configure_file (${HDF5_VOL_REST_RESOURCES_DIR}/rv_cmake_config.h.in ${HDF5_VOL_R
 #-----------------------------------------------------------------------------
 # Include the main src directory and HDF5 distribution directory
 #-----------------------------------------------------------------------------
-if (PREBUILT_HDF5_DIR)
-  set (HDF5_VOL_REST_INCLUDE_DIRECTORIES
-    ${HDF5_VOL_REST_SRC_DIR}
-    ${HDF5_VOL_REST_BINARY_DIR}
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}
-    ${PREBUILT_HDF5_DIR}/include
-    CACHE
-    INTERNAL
-    "Include directories for HDF5 REST VOL connector"
-    )
-elseif (HDF5_FOUND)
-  set (HDF5_VOL_REST_INCLUDE_DIRECTORIES
-    ${HDF5_VOL_REST_SRC_DIR}
-    ${HDF5_VOL_REST_BINARY_DIR}
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}
-    ${HDF5_INCLUDE_DIRS}
-    CACHE
-    INTERNAL
-    "Include directories for HDF5 REST VOL connector"
-  )
-else ()
-  set (HDF5_VOL_REST_INCLUDE_DIRECTORIES
+
+set (HDF5_VOL_REST_INCLUDE_DIRECTORIES
     ${HDF5_VOL_REST_SRC_DIR}
     ${HDF5_VOL_REST_BINARY_DIR}
     ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}
-    ${HDF5_DIR}
-    ${HDF5_BINARY_DIR}
-    CACHE
-    INTERNAL
+)
+
+list(APPEND HDF5_VOL_REST_INCLUDE_DIRECTORIES
+  ${HDF5_INCLUDE_DIRS}
+  ${HDF5_BINARY_DIR}/src
+  ${HDF5_SOURCE_DIR}/${HDF5_HL_DIR_NAME}/src
```

Comment:
The references to `HDF5_BINARY_DIR` and `HDF5_SOURCE_DIR` should be removed here as they shouldn't be needed and depending on how the connector is set, won't be set to anything